contracts/src/v0.8/ccip/pools/GHO/UpgradeableLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,249 @@
+
+    _transferOwnership(owner_);
+    s_router = IRouter(router);
+    if (i_allowlistEnabled) _applyAllowListUpdates(new address[](0), allowlist);
+    s_bridgeLimit = bridgeLimit;
+  }
+
+  /// @notice Locks the token in the pool
+  /// @dev The _validateLockOrBurn check is an essential security check
+  function lockOrBurn(
+    Pool.LockOrBurnInV1 calldata lockOrBurnIn
+  ) external virtual override returns (Pool.LockOrBurnOutV1 memory) {
+    // Increase bridged amount because tokens are leaving the source chain
+    if ((s_currentBridged += lockOrBurnIn.amount) > s_bridgeLimit) revert BridgeLimitExceeded(s_bridgeLimit);
+
+    _validateLockOrBurn(lockOrBurnIn);
+
+    emit Locked(msg.sender, lockOrBurnIn.amount);
+
+    return
+      Pool.LockOrBurnOutV1({
+        destTokenAddress: getRemoteToken(lockOrBurnIn.remoteChainSelector),
+        destPoolData: _encodeLocalDecimals()
+      });
+  }
+
+  /// @notice Release tokens from the pool to the recipient
+  /// @dev The _validateReleaseOrMint check is an essential security check
+  function releaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) external virtual override returns (Pool.ReleaseOrMintOutV1 memory) {
+    // This should never occur. Amount should never exceed the current bridged amount
+    if (releaseOrMintIn.amount > s_currentBridged) revert NotEnoughBridgedAmount();
+    // Reduce bridged amount because tokens are back to source chain
+    s_currentBridged -= releaseOrMintIn.amount;
+
+    _validateReleaseOrMint(releaseOrMintIn);
+
+    // Calculate the local amount
+    uint256 localAmount = _calculateLocalAmount(
+      releaseOrMintIn.amount,
+      _parseRemoteDecimals(releaseOrMintIn.sourcePoolData)
+    );
+
+    // Release to the recipient
+    getToken().safeTransfer(releaseOrMintIn.receiver, localAmount);
+
+    emit Released(msg.sender, releaseOrMintIn.receiver, localAmount);
+
+    return Pool.ReleaseOrMintOutV1({destinationAmount: localAmount});
+  }
+
+  /// @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) public pure virtual override returns (bool) {
+    return interfaceId == type(ILiquidityContainer).interfaceId || super.supportsInterface(interfaceId);
+  }
+
+  /// @notice Gets LiquidityManager, can be address(0) if none is configured.
+  /// @return The current liquidity manager.
+  function getRebalancer() external view returns (address) {
+    return s_rebalancer;
+  }
+
+  /// @notice Sets the LiquidityManager address.
+  /// @dev Only callable by the owner.
+  function setRebalancer(address rebalancer) external onlyOwner {
+    s_rebalancer = rebalancer;
+  }
+
+  /// @notice Sets the current bridged amount to other chains
+  /// @dev Only callable by the owner.
+  /// @dev Does not emit event, it is expected to only be called during token pool migrations.
+  /// @param newCurrentBridged The new bridged amount
+  function setCurrentBridgedAmount(uint256 newCurrentBridged) external onlyOwner {
```

Comment:
I'm okay with making it callable by the owner only (not the bridge limit admin), since it's something more sensitive

Reply:
yep, agreed, thoughts are alined